checker/src/main/java/org/checkerframework/checker/determinism/DeterminismQualifierPolymorphism.java

```diff
@@ -80,25 +83,58 @@ protected void replace(
                     qualHierarchy.leastUpperBounds(replacementsPolyDet, replacementsPolyDetNoOND);
             replacements = new AnnotationMirrorSet(lub);
         }
+        if (replacements == null) {
+            return;
+        }
 
         switch (value) {
             case "":
                 type.replaceAnnotations(replacements);
                 return;
             case "use":
-                if (!replacements.contains(factory.POLYDET)
-                        && !replacements.contains(factory.POLYDET_UP)
-                        && !replacements.contains(factory.POLYDET_DOWN)
-                        && !replacements.contains(factory.POLYDET_UPDET)) {
-                    type.replaceAnnotations(replacements);
+                if (replacementsPolyDet != null
+                        && replacementsPolyDet.contains(factory.POLYDET_NOORDERNONDET)) {
+                    type.replaceAnnotation(factory.POLYDET_USENOORDERNONDET);
+                }
+                // If @PolyDet("use") resolves to a qualifier that is a subtype of
```

Comment:
This comment is the English equivalent of the code.  What's "type annotation"? What is in `replacementsPolyDetUse`?